src/main/scala/beam/utils/plan/sampling/AvailableModeUtils.scala

```diff
@@ -36,16 +37,13 @@ object AvailableModeUtils extends LazyLogging {
     * @return List of excluded mode string
     */
   def getExcludedModesForPerson(population: Population, personId: String): Array[String] = {
-    Option(
+    val maybeExcludedModes = Option(
       population.getPersonAttributes.getAttribute(personId, PopulationAdjustment.EXCLUDED_MODES)
-    ).map(_.toString) match {
-      case Some(modes) =>
-        if (modes.isEmpty) {
-          Array.empty[String]
-        } else {
-          modes.split(",")
-        }
-      case None => Array.empty[String]
+    )
+    maybeExcludedModes match {
+      case Some(modes: Iterable[_])   => modes.map(_.toString).filterNot(v => isBlank(v)).toArray
```

Comment:
The split was lost here - it seems it should still be needed?

Reply:
Could be the reason for two new failures:

> 13:24:42 CarSharingSpec:
> 13:24:42 Running a car-sharing-only scenario with abundant cars
> 13:24:42 - must result in everybody driving *** FAILED *** (18 seconds, 273 milliseconds)
> 13:24:42   4 did not equal 0 Someone wasn't driving even though everybody wants to and cars abound. (CarSharingSpec.scala:147)
> 13:24:42 Running a car-sharing-only scenario with one car per person at home
> 13:24:42 - must result in everybody driving *** FAILED *** (21 seconds, 21 milliseconds)
> 13:24:42   82 did not equal 0 Someone wasn't driving even though everybody wants to and cars abound. (CarSharingSpec.scala:147)

Reply:
I put some log with the previous output and the new output. with the current change the output is the same